src/plots/plots.js

```diff
@@ -1352,21 +1368,37 @@ function calculateReservedMargins(margins) {
 }
 
 plots.supplyLayoutModuleDefaults = function(layoutIn, layoutOut, fullData, transitionData) {
+    var components = Object.keys(Registry.componentsRegistry);
+    var basePlotModules = layoutOut._basePlotModules;
     var i, _module;
 
-    // can't be be part of basePlotModules loop
-    // in order to handle the orphan axes case
-    Plotly.Axes.supplyLayoutDefaults(layoutIn, layoutOut, fullData);
+    var Cartesian = Registry.subplotsRegistry.cartesian;
+
+    // check if any components need to add more base plot modules
+    // that weren't captured by traces
+    for(i = 0; i < components.length; i++) {
+        _module = Registry.componentsRegistry[components[i]];
+
+        if(_module.includeBasePlot) {
+            _module.includeBasePlot(layoutIn, layoutOut);
+        }
+    }
+
+    // make sure we *at least* have some cartesian axes
+    if(!basePlotModules.length) {
+        basePlotModules.push(Cartesian);
+    }
+
+    // ensure all cartesian axes have at least one subplot
+    if(layoutOut._has('cartesian') || layoutOut._has('gl2d')) {
```

Comment:
Why do we need the `|| layoutOut._has('gl2d')` part here?

Reply:
In fact we don't! Good 👀 
We *depend on* gl2d inside `finalizeSubplots` but it's irrelevant if we have *only* gl2d. 6e62b35